monitoring/tests/test_collectors.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python3
+import unittest
+import requests_mock
+from datetime import datetime, timezone
+from collectors.jenkins_collector_adapter import JenkinsCollectorAdapter, InvalidResponseCodeError
+from stats import BuildStats, StageStats
+
+
+class TestJenkinsGetBuildStatsEndpoint(unittest.TestCase):
+    def test_build_stats_endpoint_is_good(self):
+        url = JenkinsCollectorAdapter.get_build_stats_endpoint(build_url='http://localhost:8080/job/MyJob/1')
+        self.assertEqual(url, 'http://localhost:8080/job/MyJob/1/wfapi/describe')
+
+
+class TestJenkinsGetRawStats(unittest.TestCase):
+    def test_get_raw_stats_ok(self):
+        with requests_mock.mock() as m:
+            m.get('http://localhost:8080/job/MyJob/1/wfapi/describe', json={
```

Comment:
This is really minor, and Dariusz disagrees, but IMO we shouldn't use a "realistic" URL like `localhost` and port `8080` as test strings. It may be confusing. I would suggest something like `asdf:5678`. This would need to be changed throughout all files.